test/integration/src/test/java/io/pravega/test/integration/RestoreBackUpDataRecoveryTest.java

```diff
@@ -546,6 +582,428 @@ private void readEventsFromStreams(ClientFactoryImpl clientFactory, ReaderGroupM
+        }
+
+        assertNotNull(currentTimeWindow.getUpperTimeBound());
+
+        currentTimeWindow = verifyEventsWithTimeBounds(streamObj, reader, event, currentTimeWindow);
+
+        assertNotNull(currentTimeWindow.getLowerTimeBound());
+    }
+
+    /**
+     * Gets watermarks used while writing the events
+     */
+    private LinkedBlockingQueue<Watermark> getWatermarks(PravegaRunner pravegaRunner, AtomicBoolean stopFlag,
+                                                         CompletableFuture<Void> writer1Future, CompletableFuture<Void>
+                                                                 writer2Future) throws Exception {
+        @Cleanup
+        SynchronizerClientFactory syncClientFactory = SynchronizerClientFactory.withScope(SCOPE,
+                ClientConfig.builder().controllerURI(pravegaRunner.controllerRunner.controllerURI).build());
+
+        String markStream = NameUtils.getMarkStreamForStream(STREAM1);
+        RevisionedStreamClient<Watermark> watermarkReader = syncClientFactory.createRevisionedStreamClient(markStream,
+                new WatermarkSerializer(), SynchronizerConfig.builder().build());
+        LinkedBlockingQueue<Watermark> watermarks = new LinkedBlockingQueue<>();
+        fetchWatermarks(watermarkReader, watermarks, stopFlag);
+        AssertExtensions.assertEventuallyEquals(true, () -> watermarks.size() >= 2, 100000);
+        stopFlag.set(true);
+        writer1Future.join();
+        writer2Future.join();
+        return watermarks;
+    }
+
+    private List<Map<Stream, StreamCut>> getStreamCutsFromWaterMarks(Stream streamObj, LinkedBlockingQueue<Watermark> watermarks)
+            throws InterruptedException {
+        Watermark watermark0 = watermarks.take();
+        Watermark watermark1 = watermarks.take();
+        assertTrue(watermark0.getLowerTimeBound() <= watermark0.getUpperTimeBound());
+        assertTrue(watermark1.getLowerTimeBound() <= watermark1.getUpperTimeBound());
+        assertTrue(watermark0.getLowerTimeBound() < watermark1.getLowerTimeBound());
+
+        Map<Segment, Long> positionMap0 = watermark0.getStreamCut().entrySet().stream().collect(
+                Collectors.toMap(x -> new Segment(SCOPE, STREAM1, x.getKey().getSegmentId()), Map.Entry::getValue));
+        Map<Segment, Long> positionMap1 = watermark1.getStreamCut().entrySet().stream().collect(
+                Collectors.toMap(x -> new Segment(SCOPE, STREAM1, x.getKey().getSegmentId()), Map.Entry::getValue));
+
+        StreamCut streamCutFirst = new StreamCutImpl(streamObj, positionMap0);
+        StreamCut streamCutSecond = new StreamCutImpl(streamObj, positionMap1);
+        Map<Stream, StreamCut> firstMarkStreamCut = Collections.singletonMap(streamObj, streamCutFirst);
+        Map<Stream, StreamCut> secondMarkStreamCut = Collections.singletonMap(streamObj, streamCutSecond);
+
+        return new ArrayList<Map<Stream, StreamCut>>(Arrays.asList(firstMarkStreamCut, secondMarkStreamCut));
```

Comment:
Why do you re-wrap a list as a new list?

Reply:
Fixed.